catkit/hardware/mcpherson/scan_controller_789a4.py

```diff
@@ -0,0 +1,616 @@
+        :param wait: bool (optional) - Whether to wait for motion to have stopped before returning from this func.
+
+        :return: int - steps_per_second, actual motor velocity.
+        """
+
+        cmd = ASCIIControlCodes.SCAN_UP if steps >= 0 else ASCIIControlCodes.SCAN_DOWN
+
+        initial_scan_speed = actual_scan_speed = self.read_params()[Parameters.SCAN_SPEED]
+        velocity_changed = False
+
+        # Set speed.
+        if steps_per_second is not None:
+            if steps_per_second != actual_scan_speed:
+                actual_scan_speed = self.set_slew_speed(steps_per_second)
+                velocity_changed = True
+
+        self.command(cmd, steps)
+
+        if wait:
+            self.await_stop()
+
+        if reset_speed and velocity_changed:
+            self.set_slew_speed(initial_scan_speed)
+
+        return actual_scan_speed
+
+    def home(self):
+        """ This is implemented by the derived class below. """
+        raise NotImplementedError("This model does not support limit switches and, therefore, has no homing feature.")
+
+
+class McPherson789A4WithLimitSwitches(McPherson789A4):
+    _HAS_LIMIT_SWITCHES = True
+    DEFAULT_HOMING_TIMEOUT = 60 * 5  # Depending on the initial motor position and velocity, this may take a while.
+    DEFAULT_HOMING_SPEED = 23000  # As documented in the manual.
+    FAST_HOMING_SPEED = 3000000  # As documented in the manual.
+    EDGE_DETECTION_SPEED = 1000  # As documented in the manual.
+
+    def initialize(self, visa_id, steps_per_rev=36000, timeout=1, home_on_startup=True):
+        super().initialize(visa_id, steps_per_rev=steps_per_rev, timeout=timeout)
+
+        self.home_on_startup = home_on_startup
+        if not self.home_on_startup:
+            warnings.warn("It is strongly recommended to home on startup before proceeding with anything else!")
+
+    def _open(self, *args, **kwargs):
+        self.instrument = super()._open(*args, **kwargs)
+
+        if self.home_on_startup:
+            self.home()
+
+        return self.instrument
+
+    def get_limit_status(self):
+        """ Query the device for limit status.
+
+        :return: LimitSwitchStatus.LOW, LimitSwitchStatus.HIGH, LimitSwitchStatus.NO_LIMIT. NOTE: This does not return
+                 LimitSwitchStatus.HOME for that equivalent functionality call self.is_home().
+        """
+        resp = np.int32(self.command(ASCIIControlCodes.GET_LIMIT_SWITCH_STATUS))
+
+        is_home = resp & np.int32(LimitSwitchStatus.HOME.value)
+
+        if is_home:
+            resp = resp - np.int32(LimitSwitchStatus.HOME.value)
+
+        status = LimitSwitchStatus(resp)
+
+        return status
+
+    def poll_limit_status(self, break_conditions, *args, **kwargs):
+        """ Poll device for limit status. See self.poll_status() for more info. """
+        return self.poll_status(break_conditions, *args, **kwargs)
```

Comment:
This isn't used and could be deleted. However, if kept, it's currently wrong and should be:
```python
return self.poll_status(break_conditions, self.getlimit_status, *args, **kwargs)
```